packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/dataStreamSearch/workspaceSelector/workspaceSelector.tsx

```diff
@@ -16,16 +18,30 @@ export interface WorkspaceSelectorProps {
   control: Control<SearchFields>;
   client: IoTTwinMakerClient;
   OnGettingError: (isError: boolean) => void;
+  setValue: UseFormSetValue<SearchFields>;
 }
 
-export const WorkspaceSelector = ({ control, client, OnGettingError }: WorkspaceSelectorProps) => {
+export const WorkspaceSelector = ({ control, client, OnGettingError, setValue }: WorkspaceSelectorProps) => {
   const { workspaces, status } = useWorkspaces({ client });
   const workspaceOptions = createWorkspaceOptions(workspaces);
 
+  const [selectedWorkspace, setSelectedWorkspace] = useLocalStorage<OptionDefinition | null>('selectedWorkspace', null);
```

Comment:
instead of using null, can it be undefined ?

Reply:
We are setting null as defaultValue for 'workspace' form field, so using the same type for localStorage as well
![image](https://github.com/awslabs/iot-app-kit/assets/154328164/a46c7fac-fa79-4ec9-accf-f5e56144fa67)

Reply:
No, null is used as defaultValue for workspace form field so we have used the same
![image](https://github.com/awslabs/iot-app-kit/assets/154328164/03d4d36e-9770-463e-9c35-e7cb6b34556f)